### PR TITLE
[PPP-4223] - Use of vulnerable component spring-core before 4.3.14, s…

### PR DESCRIPTION
**Pull request by ricardosilva88**
**DO NOT MERGE WITHOUT THE REST OF PRs:**
[pentaho/maven-parent-poms#95 (comment)](https://github.com/pentaho/maven-parent-poms/pull/95#issue-250486125)

### CODE REVIEW

File: api/pom.xml [hunk 1]
```diff
@@ -51,11 +51,11 @@
       <artifactId>dom4j</artifactId>
       <scope>compile</scope>
     </dependency>
+
+    <!-- Needed for runtime testing -->
     <dependency>
       <groupId>org.springframework.security</groupId>
       <artifactId>spring-security-core</artifactId>
-      <version>${spring-security.version}</version>  <!-- see maven-parent-pom -->
-      <scope>compile</scope>
```
Comment (graimundo) [inline api/pom.xml]: If this artifact is just needed during tests (as mentioned in the comment https://github.com/pentaho/pentaho-platform/pull/4341/files#diff-8658bb96fff6ac848cd28d3242fce067R55 ) then scope should be test `<scope>test</scope>`.

File: api/pom.xml [hunk 2]
```diff
@@ -66,15 +66,15 @@
     <dependency>
       <groupId>org.springframework</groupId>
       <artifactId>spring-beans</artifactId>
-      <version>${spring.framework.version}</version>
-      <scope>compile</scope>
```
Comment (graimundo) [inline api/pom.xml]: If this artifact is just needed during tests (as mentioned in the comment https://github.com/pentaho/pentaho-platform/pull/4341/files#diff-8658bb96fff6ac848cd28d3242fce067R55 ) then scope should be test `<scope>test</scope>`.

File: assemblies/pentaho-war/pom.xml [hunk 3]
```diff
@@ -50,6 +50,13 @@
     </dependency>
     <!-- endregion -->
 
+    <!-- Needed for runtime -->
+    <dependency>
+      <groupId>org.springframework</groupId>
+      <artifactId>spring-context-support</artifactId>
```
Comment (graimundo) [inline assemblies/pentaho-war/pom.xml]: If this artifact is needed at runtime then scope should be runtime `<scope>runtime</scope>`.
This artifact is for runtime correct? Not tests?

File: core/pom.xml [hunk 4]
```diff
@@ -296,21 +285,25 @@
       </exclusions>
     </dependency>
     <dependency>
-      <groupId>org.springframework.security</groupId>
-      <artifactId>spring-security-acl</artifactId>
-      <version>${spring-security.version}</version>  <!-- see maven-parent-pom -->
+      <groupId>org.springframework</groupId>
+      <artifactId>spring-aop</artifactId>
       <scope>compile</scope>
```
Comment (graimundo) [inline core/pom.xml]: If this artifact is just needed during tests then scope should be test `<scope>test</scope>`.

File: core/pom.xml [hunk 5]
```diff
@@ -296,21 +285,25 @@
...
+    </dependency>
+    <dependency>
+      <groupId>org.springframework.security</groupId>
+      <artifactId>spring-security-core</artifactId>
       <exclusions>
```
Comment (graimundo) [inline core/pom.xml]: If this artifact is just needed during tests then scope should be test `<scope>test</scope>`.

File: core/pom.xml [hunk 6]
```diff
@@ -296,21 +285,25 @@
...
         <exclusion>
           <artifactId>*</artifactId>
           <groupId>*</groupId>
         </exclusion>
       </exclusions>
     </dependency>
+    <!-- -->
```
Comment (graimundo) [inline core/pom.xml]: I'm assuming this `<!-- -->`marker, marks the end of `test` scope dependencies.
You can use Intellij Idea regions to integrate better with our development tools: https://youtrack.jetbrains.com/issue/IDEA-93649
Just use `<region RegionName -->` and `<!-- endregion -->` to have folding directly available in the IDE.

File: api/pom.xml (hunk 1 above)
Comment (ricardosilva88) [reply api/pom.xml]: Nop, also needed for compilation...just checked

File: api/pom.xml (hunk 2 above)
Comment (ricardosilva88) [reply api/pom.xml]: Nop, also needed for compilation...just checked

File: assemblies/pentaho-war/pom.xml (hunk 3 above)
Comment (ricardosilva88) [reply assemblies/pentaho-war/pom.xml]: Yeap, just confirmed. I'll change it

File: core/pom.xml (hunk 6 above)
Comment (ricardosilva88) [reply core/pom.xml]: It makes sense, I will change in all PR's to follow this format.

**Comment by ricardosilva88** [comment]
Updated PR

File: core/pom.xml (hunk 4 above)
Comment (ricardosilva88) [reply core/pom.xml]: Needed also for compilation, removed commentary

File: core/pom.xml (hunk 5 above)
Comment (ricardosilva88) [reply core/pom.xml]: Needed also for compilation, removed commentary

File: api/pom.xml (hunk 1 above)
Comment (ricardosilva88) [reply api/pom.xml]: If I put <scope>test</scope> it doesn't compile

**Pull request merged**